lib/Backends/CPU/CPUDeviceManager.h

```diff
@@ -13,13 +13,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#ifndef GLOW_BACKENDS_CPUDEVICEMANAGER_H
-#define GLOW_BACKENDS_CPUDEVICEMANAGER_H
+#ifndef GLOW_BACKENDS_CPU_CPUDEVICEMANAGER_H
+#define GLOW_BACKENDS_CPU_CPUDEVICEMANAGER_H
 
 #include "glow/Backends/QueueBackedDeviceManager.h"
 
 namespace glow {
 
+/// A class controlling a single CPU thread of execution driving the JIT
+/// backend. Many CPUFunctions may be added, but only one inference is executed
```

Comment:
would be interesting to check what stops real parallel execution on CPU backend (given we have separate addresses for inputs/outputs through per inference context, as well as per inference allocation of activations).

Reply:
I'm pretty sure nothing stops it, and we do have real parallel execution in the resnet-concurrent example (https://github.com/pytorch/glow/blob/master/examples/resnet-concurrent.cpp).

I do think that a multihreaded CPUDeviceManager is less useful than a single threaded one because it's not a good proxy for the accelerator backed DeviceManagers. A single thread of execution is an easy tool for the Provisioner/Executor to use without needing to know about the details.